### ISSUE TITLE
Consider changing the ThreadCount on OS X to 1

### ISSUE THREAD

**Issue by halter73** [aspnet/KestrelHttpServer#921]
`ThreadCountTests` have been [pretty](https://github.com/aspnet/KestrelHttpServer/pull/908) [flaky](https://github.com/aspnet/KestrelHttpServer/pull/913) on OS X. It now seems we are working around this issue by always running Kestrel tests with a `ThreadCount` of 1 on OS X.

The working assumption is that the test failures are caused by exceeding the file open limit, but output from `lsof` seems to contradict this.

Since we have no fix for this issue other than changing the tests not use multiple thread, I really think we should consider not having Kestrel run in an untested configuration on OS X. I think this would mean disallowing `ThreadCount`s other than 1 on OS X or at least making 1 the default.

@cesarbs @muratg

**Comment by muratg** [comment]
@mikeharder @davidfowl Thoughts?  This would fix a lot of flakiness that people may see on OS X.

**Comment by cesarbs** [comment]
For people running a single app in development, `ulimit`, `launchctl`, etc. fix the flakiness. We shouldn't make a decision like this based on a flaky test whose flakiness is not due to product issues.

**Comment by halter73** [comment]
@cesarbs How sure are you of this? It's not like we often test Kestrel on OS X with a bunch of concurrent requests. The closest we get is probably the failing test.

**Comment by cesarbs** [comment]
@halter73 I did the following: I ran the sample app with 8 threads on a shell using the default ulimit (256). On another shell I issued 100 concurrent requests against the app and saw the errors we were seeing in the tests. Then I increased `ulimit -n` to 2048 in the shell where I was running the app, issues the requests again (multiple times) and never saw the errors again.

Restricting thread count on OS X is too aggressive. If people hit issues we can tell them to 1) increase the ulimit or 2) set the thread count to 1 themselves.
